tezos/context/src/kv_store/in_memory.rs

```diff
@@ -209,7 +219,36 @@ impl KeyValueStoreBackend for InMemory {
     }
 
     fn memory_usage(&self) -> RepositoryMemoryUsage {
-        self.hashes.get_memory_usage()
+        let mut mem = self.hashes.get_memory_usage();
+        mem.nshapes = self.shapes.nshapes();
+        mem
+    }
+
+    fn get_shape(&self, shape_id: DirectoryShapeId) -> Result<ShapeStrings, DBError> {
+        self.shapes
+            .get_shape(shape_id)
+            .map(ShapeStrings::SliceIds)
+            .map_err(Into::into)
+    }
+
+    fn make_shape(
+        &mut self,
+        dir: &[(StringId, DirEntryId)],
+    ) -> Result<Option<DirectoryShapeId>, DBError> {
+        self.shapes.make_shape(dir).map_err(Into::into)
+    }
+
+    fn update_strings(&mut self, string_interner: &StringInterner) -> Result<(), DBError> {
+        // Clone it only if it is has changed
+        if string_interner != &self.string_interner {
+            self.string_interner = string_interner.clone();
```

Comment:
@sebastiencs how about the `push_str` here to avoid the clone, can that be done? And also just compare the length instead? And then the slice resulting from the difference between the two lengths is pushed.